rocrate_validator/profiles/five-safes-crate/should/10_outputs.ttl

```diff
@@ -0,0 +1,91 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@prefix ro: <./> .
+@prefix ro-crate: <https://github.com/crs4/rocrate-validator/profiles/ro-crate/> .
+@prefix five-safes-crate: <https://github.com/eScienceLab/rocrate-validator/profiles/five-safes-crate/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix schema: <http://schema.org/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix validator: <https://github.com/crs4/rocrate-validator/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+
+five-safes-crate:CreateActionHasResultIfActionCompleted
+  a sh:NodeShape ;
+  sh:name "CreateAction" ;
+  sh:description "`CreateAction` with CompleteActionStatus SHOULD have the `schema:result` property." ;
+
+
+  sh:target [
+        a sh:SPARQLTarget ;
+        sh:prefixes ro-crate:sparqlPrefixes ;
+        sh:select """
+            SELECT ?this WHERE {
+              ?this a schema:CreateAction ;
+                      schema:actionStatus "http://schema.org/CompleteActionStatus" .
+            }
+        """
+    ] ;
+
+
+  sh:property [
+    a sh:PropertyShape ;
+    sh:name "Result" ;
+    sh:path schema:result ;
+    sh:minCount 1 ;
+    sh:severity sh:Warning ;
+    sh:message "`CreateAction` with CompleteActionStatus SHOULD have the `schema:result` property." ;
+  ] .
+
+
+
+# This test passes even when it should fail, so it is disabled for now.
+# According to Clause.ai the possible reason for this misbehaviour is that
+# the RO-Crate validator adds inferred types like schema:MediaObject to the
+# entity based on other characteristics (like it being a file). So the entity
+# has BOTH "Filexx" (which doesn't expand to a schema.org URI) AND the inferred
+#  schema:MediaObject, which makes the constraint pass.
+# The fundamental problem is that we can't easily distinguish between
+# user-declared types and validator-inferred types in the materialized RDF graph.
+
+# five-safes-crate:CreateActionResultOutputsHaveAllowedTypes
+#   a sh:NodeShape ;
+#     sh:name "Output" ;
+#     sh:target [
+#       a sh:SPARQLTarget ;
+#       sh:select """
+#         PREFIX schema: <http://schema.org/>
+#         SELECT ?this
+#         WHERE {
+#           ?createAction a schema:CreateAction .
+#           ?createAction schema:result ?this .
+#         }
+#       """ ;
+#     ] ;
+#     sh:message "Every schema:result must point to a defined node (must have an rdf:type)." ;
+#     sh:property [
+#       sh:path rdf:type ;
+#       sh:qualifiedValueShape (
+#         schema:MediaObject
+#         schema:Dataset
+#         schema:Collection
+#         schema:DigitalDocument
+#         schema:PropertyValue
+#       ) ;
+#       sh:qualifiedMinCount 1 ;
+#       sh:message "Result has a rdf:type outside the allowed set." ;
+#       sh:severity sh:Warning ;
+#     ] .
```

Comment:
Instead of using `sh:property` to target the type, it's better to use `sh:class` on the NodeShape (as in [these examples](https://book.validatingrdf.com/bookHtml011.html#sec131))
```suggestion
  sh:or (
      [
        sh:class schema:MediaObject;
      ]
      [
        sh:class schema:Dataset;
      ]
      [
        sh:class schema:Collection;
      ]
      [
        sh:class schema:DigitalDocument;
      ]
      [
        sh:class schema:PropertyValue;
      ]
    );
```
This will allow target nodes to have additional classes too.